create-db-ddl.sql

```diff
@@ -1206,135 +1207,24 @@ CREATE TABLE IF NOT EXISTS cqc."UserAudit" (
-ALTER TABLE cqc."User" ADD COLUMN "EmailChangedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "EmailSavedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "EmailChangedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "PhoneSavedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "PhoneChangedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "PhoneSavedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "PhoneChangedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionSavedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionChangedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionSavedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionChangedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionAnswerSavedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionAnswerChangedAt" TIMESTAMP NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionAnswerSavedBy" VARCHAR(120) NULL;
-ALTER TABLE cqc."User" ADD COLUMN "SecurityQuestionAnswerChangedBy" VARCHAR(120) NULL;
-
-
--- add the created/updated/updatedBy columns
-ALTER TABLE cqc."User" ADD COLUMN created TIMESTAMP WITHOUT TIME ZONE NOT NULL DEFAULT NOW();
-ALTER TABLE cqc."User" ADD COLUMN updated TIMESTAMP WITHOUT TIME ZONE NOT NULL DEFAULT NOW();
-ALTER TABLE cqc."User" ADD COLUMN updatedby VARCHAR(120) NULL;
-UPDATE cqc."User" set updatedby='admin';                            -- cannot be null, so setting a default value on apply patch
-ALTER TABLE cqc."User" ALTER COLUMN updatedby SET NOT NULL;
-
--- and drop the now unused "DateCreated" column
-ALTER TABLE cqc."User" DROP COLUMN "DateCreated";
-
-CREATE TYPE cqc."UserAuditChangeType" AS ENUM (
-    'created',
-    'updated',
-    'saved',
-    'changed',
-    'passwdReset',
-    'loginSuccess',
-    'loginFailed',
-    'loginWhileLocked'
-);
-CREATE TABLE IF NOT EXISTS cqc."UserAudit" (
-	"ID" SERIAL NOT NULL PRIMARY KEY,
-	"UserFK" INTEGER NOT NULL,
-	"Username" VARCHAR(120) NOT NULL,
-	"When" TIMESTAMP WITHOUT TIME ZONE NOT NULL DEFAULT NOW(),
-	"EventType" cqc."UserAuditChangeType" NOT NULL,
-	"PropertyName" VARCHAR(100) NULL,
-	"ChangeEvents" JSONB NULL,
-	CONSTRAINT "WorkerAudit_User_fk" FOREIGN KEY ("UserFK") REFERENCES cqc."User" ("RegistrationID") MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION
-);
-CREATE INDEX "UserAudit_UserFK" on cqc."UserAudit" ("UserFK");
-
--- https://trello.com/c/1f4RSnlu defect fix
-DROP INDEX IF EXISTS cqc."Establishment_unique_registration";
-DROP INDEX IF EXISTS cqc."Establishment_unique_registration_with_locationid";
-CREATE UNIQUE INDEX IF NOT EXISTS "Establishment_unique_registration" ON cqc."Establishment" ("Name", "PostCode", "LocationID");
-CREATE UNIQUE INDEX IF NOT EXISTS "Establishment_unique_registration_with_locationid" ON cqc."Establishment" ("Name", "PostCode") WHERE "LocationID" IS NULL;
+	"UserUID" = "USER_UUID"."UIDv4"
```

Comment:
Hi Warren

I didn't get  "UserUID" = "USER_UUID"."UIDv4"  and then used from word. We might need to discuss this.


Thanks

Shakir

Reply:
Sorry It was my mistake found complete code it was just confusion from me only.

--  update existing User records with UUID
UPDATE
	cqc."User" 
SET
	"UserUID" = "USER_UUID"."UIDv4"
FROM (
	SELECT CAST(substr(CAST(myuuids."UID" AS TEXT), 0, 15) || '4' || substr(CAST(myuuids."UID" AS TEXT), 16, 3) || '-89' || substr(CAST(myuuids."UID" AS TEXT), 22, 36) AS UUID) "UIDv4", "RegID"
    FROM (
        SELECT uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) "UID",
                "User"."RegistrationID" "RegID"
        FROM cqc."User", cqc."Login"
        WHERE "User"."RegistrationID" = "Login"."RegistrationID"
	) AS MyUUIDs
) AS "USER_UUID"
WHERE "USER_UUID"."RegID" = "User"."RegistrationID";

ALTER TABLE cqc."User" ALTER COLUMN "UserUID" SET NOT NULL;